lib/active_model_serializers/adapter/json_api.rb

```diff
@@ -22,11 +22,12 @@ def initialize(serializer, options = {})
       # {http://jsonapi.org/format/#document-top-level data and errors MUST NOT coexist in the same document.}
       def serializable_hash(options = nil)
         options ||= {}
-        if serializer.success?
-          success_document(options)
-        else
-          failure_document
-        end
+        document = if serializer.success?
+                     success_document(options)
+                   else
+                     failure_document
+                   end
+        translate_key_casing!(document, options[:serialization_context])
```

Comment:
is this method mutating? or does it return a different hash?

like,

does `document.object_id == translate_key_casing!(document, {...}).object_id`?